src/models/imposter.js

```diff
@@ -54,6 +54,9 @@ function create (Protocol, request) {
 
                 result.requests = server.requests;
                 result.stubs = server.stubs();
+                result.stubs.forEach(function (stub) {
```

Comment:
Yep - initializing the state object when creating the protocols feels right, and you inject it into the stubs in the call to the stubRepository (e.g. line 59 in tcpServer and similar lines in other protocols). That approach is enough to make all your tests pass, except the imposterTest unit tests, which can simply revert back to what they were for all the tests that assert against the stubs, and if desired add a test that validates the state shows at the root level. Most importantly, the actual behavior of allowing a response injection to change the state and a predicate injection to use it will continue to work.

The reason it's important to change this line to show the state at the root level instead of within the stubs is because a developer who's troubleshooting why their predicate isn't working may want to see the stub. If we present it in the JSON as if it were under each stub, that would create the false impression that they can change it stub-by-stub. Showing it at the root correctly shows it as shared between the stubs. Since it's API visible, it's not something I can change later without making it a breaking change, which I'm very hesitant to do.